cmd/gaia/app/app.go

```diff
@@ -229,7 +248,9 @@ func (app *GaiaApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.R
 	validatorUpdates, endBlockerTags := staking.EndBlocker(ctx, app.stakingKeeper)
 	tags = append(tags, endBlockerTags...)
 
-	app.assertRuntimeInvariants()
+	if app.assertInvariantsBlockly {
+		app.assertRuntimeInvariants()
```

Comment:
Hmm, I don't know if this is quite the right option. Can we instead have an automated way to scan for an invariant break and trigger a script to submit the invariant-break tx or something?

Reply:
Okay so the idea by introducing this was to create a way for special full nodes to run with the invariants which would then panic first - after they could manually create the required transaction. I think it makes sense to keep this as an optional full node operation and _not_ as a mandatory parallel process on all nodes (is that what you were thinking?). 

We could further develop a more advanced system whereby with a configurable hot key the node could automatically generate, sign, and broadcast the relevant transaction - this sounds cool. Thinking out loud is there maybe even a way where this wouldn't need to be a hot key, but of course the transaction-constant-fee funds would still need to be locked up when this special node was initiated? Thoughts?

Reply:
> create a way for special full nodes to run with the invariants which would then panic first - after they could manually create the required transaction

Right, I agree with this - it just seems like panicking without further information wouldn't work, since the full node had better make sure that the panic was due to an invariant break before it spends 1000 Atoms on a invariant transaction.

> We could further develop a more advanced system whereby with a configurable hot key the node could automatically generate, sign, and broadcast the relevant transaction - this sounds cool.

That's definitely in line with what I'm thinking - I think it's OK to require a hot key; a node operator who wishes to monitor in this fashion should just send the deposit to a separate hot-key account, which will automatically sign an invariant-break tx if an invariant is broken. If you have a solution that doesn't require a hot key, that could be interesting too - how would that work?

Reply:
Okay so there are a couple easy options here. 
First we could simple print out the required transaction to run under the existing process flow as a part of the panic message (easy). Next we could implement the automatic broadcast of that transaction. Alternative to using a hotkey we could simply pre-sign all the relevant messages which could be broadcast under the panic situation (only 1 of them ever would however). I think this even has slightly better security principals over a hot key as if the server got hacked, the atoms could only be burned through sending the message (and not directly stolen)

Reply:
I don't think it's important to have immediately; I think we should just provision for its future implementation and leave a note about it.

Reply:
k I'll open an issue

Reply:
K prior implemented in https://github.com/cosmos/cosmos-sdk/pull/3656/commits/2d79e248e1b072b352c56d24f2c2faa48d6c113b
new issue here https://github.com/cosmos/cosmos-sdk/issues/3969